src/server/rdb_save.cc

```diff
@@ -240,6 +240,10 @@ std::error_code RdbSerializer::SaveValue(const PrimeValue& pv) {
         ec = SaveString(pv.GetSlice(&tmp_str_));
       }
     }
+    // We flush here because if the next element in the bucket we are serializing is a container,
+    // it will first serialize the first entry and then flush the internal buffer, even if
+    // crossed the limit.
+    FlushIfNeeded(FlushState::kFlushEndEntry);
```

Comment:
@adiholden This is a general issue not only limited to strings right ? Same applies for other data types as well that are not chunked (treated as big values)? So for example, `we don't` flush after we serialize a `json`. What if, all the elements in the bucket are json data types ? Shall I take care of that as well.

Reply:
why here and not in RdbSerializer::SaveEntry at the end of the function? this way you can be sure its actually after you finished saving the entry and not in the middle if you have some other serialization after SaveValue which I see we dont have today but might in the future. By moving to the end of save entry it will apply also to all datatypes

Reply:
p.s add a test